src/Atc.Cosmos/DependencyInjection/CosmosBuilder{T}.cs

```diff
@@ -0,0 +1,28 @@
+using Atc.Cosmos.Internal;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+
+namespace Atc.Cosmos.DependencyInjection
+{
+    public class CosmosBuilder<T> : CosmosBuilder, ICosmosBuilder<T>
+        where T : class, ICosmosResource
+    {
+        public CosmosBuilder(
+            IServiceCollection services)
+            : base(services)
+        {
+        }
+
+        public ICosmosBuilder<T> WithChangeFeedProcessor<TProcessor>()
+            where TProcessor : class, IChangeFeedProcessor<T>
+        {
+            Services.TryAddSingleton<ICosmosContainerInitializer, LeasesContainerInitializer>();
+            Services.TryAddSingleton<IChangeFeedFactory, ChangeFeedFactory>();
+            Services.AddSingleton<TProcessor>();
+
+            Services.AddHostedService<ChangeFeedService<T, TProcessor>>();
```

Comment:
How do you ensure  that all `ICosmosContainerInitializer`'s are completed before the hosted service is started?

> If they don't the hosted service will fail and never start.

Reply:
Good point. Will fix